src/hotspot/share/opto/output.cpp

```diff
@@ -1790,8 +1790,11 @@ void PhaseOutput::fill_buffer(C2_MacroAssembler* masm, uint* blk_starts) {
     if (C->failing()) {
       return; // CodeBuffer::expand failed
     }
-    // Emit the deopt handler code.
-    _code_offsets.set_value(CodeOffsets::Deopt, HandlerImpl::emit_deopt_handler(masm));
+    // Emit the deopt handler code if needed.
+    if (AlwaysEmitDeoptStubCode
```

Comment:
I count 5 times this same expression is repeated.  Maybe it would make sense to move it to an inlined helper function.